Kickstarter-iOS/Views/Controllers/RewardsCollectionViewController.swift

```diff
@@ -73,18 +73,9 @@ final class RewardsCollectionViewController: UICollectionViewController {
 
     guard let layout = self.flowLayout else { return }
```

Comment:
In an ideal scenario...this `guard` is repeated twice in the flow...first time here...second time in `updateHiddenScrollViewBoundsIfNeeded`. In order to make each function independent of the state of the view controller could we do this

```swift
// Callsite
guard let layout = self.flowLayout else { return }

// or maybe even also injecting layout to a new function and extracting this to `updateItemSize(for: layout)`
layout.itemSize = self.calculateItemSize(from: layout, using: self.collectionView)

self.updateHiddenScrollViewBoundsIfNeeded(for: layout)

// Updated function input
private func updateHiddenScrollViewBoundsIfNeeded(for layout: UICollectionViewFlowLayout) {
  // Then we don't need to guard here
}

// Possibly new function
private func updateItemSize(for layout: UICollectionViewFlowLayout) {
  // Then we don't need to guard here
}
```

or even something else?

Reply:
I like it, will update the `updateHiddenScrollViewBoundsIfNeeded` to accept `layout` as an argument 👍